src/twisted/web/server.py

```diff
@@ -878,3 +884,110 @@ def acceptableProtocols(self):
             baseProtocols.insert(0, b"h2")
 
         return baseProtocols
+
+
+class _HTTPSRedirectProtocolWrapper(ProtocolWrapper):
+    """
+    Wraps the TLS protocol and redirect to HTTPS if an HTTP request is
+    detected after a connection is made.
+    """
+
+    def dataReceived(self, data):
+        """
+        Only called when initial connection data is received over SSL.
+
+        After that, the dataReceived method of the wrapped protocol is called.
+
+        It checks if an HTTP request is made over HTTPS port, instead of an
+        SSL handshake.
+
+        @param data: The data received from the remote HTTP client peer.
+        @type data: L{bytes}.
+        """
+
+        def get_http_host(data):
+            """
+            Return the value of the Host header found inside `data` HTTP
+            raw request.
+
+            @param data: The first data chunk received from the remote HTTP client peer.
+            @type data: L{bytes}.
+
+            @return: The value of the Host header. It returns the empty string when the value could not be extracted.
+            @rtype: L{bytes}
+            """
```

Comment:
I feel like this level of docstring detail is a bit overkill in a callback (nested callables like this don't need docstrings per the coding standard)

Reply:
I will remove the docstring.

Regarding the name... I thought that getHTTPHost is  only for methods ...and functions names are with underscore.

Will review the coding style and update future patches.

Thanks